lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1959,7 +1959,9 @@ void TypeChecker::validateDecl(OperatorDecl *OD) {
 llvm::Expected<SelfAccessKind>
 SelfAccessKindRequest::evaluate(Evaluator &evaluator, FuncDecl *FD) const {
   if (FD->getAttrs().getAttribute<MutatingAttr>(true)) {
-    if (!FD->isInstanceMember() || !FD->getDeclContext()->hasValueSemantics()) {
+    auto functionDC = FD->getDeclContext();
+    if (!FD->isInstanceMember() ||
+        (functionDC->isTypeContext() && !functionDC->hasValueSemantics())) {
```

Comment:
`FD->isInstanceMember()` already implies `functionDC->isTypeContext()`, no? (here and below)

Reply:
Hmm I am not sure, it doesn't check for a type context internally (but for a context name and the index of the Self parameter). Let me verify

Reply:
That's the `isInstanceMember` in Parser. The one in ValueDecl does start off by checking for a type context.

Reply:
Oops, you're right. Xcode took me to the wrong place and I didn't notice. I'll remove the extra check now.

Reply:
Done.